1-Authentication/2-sign-in-b2c/App/authPopup.js

```diff
@@ -2,13 +2,74 @@
 // configuration parameters are located at authConfig.js
 const myMSALObj = new msal.PublicClientApplication(msalConfig);
 
-let accountId = "";
-let username = "";
+let accountId = '';
+let username = '';
+
+myMSALObj.addEventCallback((event) => {
+    console.log(event.eventType);
+    if (
+        (event.eventType === msal.EventType.LOGIN_SUCCESS ||
+            event.eventType === msal.EventType.ACQUIRE_TOKEN_SUCCESS) &&
+        event.payload.account
+    ) {
+        /**
+         * For the purpose of setting an active account for UI update, we want to consider only the auth
+         * response resulting from SUSI flow. "tfp" claim in the id token tells us the policy (NOTE: legacy
+         * policies may use "acr" instead of "tfp"). To learn more about B2C tokens, visit:
+         * https://docs.microsoft.com/en-us/azure/active-directory-b2c/tokens-overview
+         */
+
+        if (event.payload.idTokenClaims['tfp'] === b2cPolicies.names.editProfile) {
+            const originalSignInAccount = myMSALObj
+                .getAllAccounts()
+                .find(
+                    (account) =>
+                        account.idTokenClaims.oid === event.payload.idTokenClaims.oid &&
```

Comment:
what happens if/when oid claim is not available?

Reply:
Updated!.